lib/assets/csv_script.rb

```diff
@@ -0,0 +1,64 @@
+require 'roo'
+require 'csv'
+
+XLSX_PATH = ARGV[0] || "lib/assets/initial-list.xlsx"
+CSV_PATH = ARGV[1] || "lib/assets/pre-chunk-content.csv"
+NEW_CSV_ROOT = ARGV[2] || "lib/assets/2023-01-12-pilot-SNAP-warning-notice_"
+
+xlsx = Roo::Excelx.new(XLSX_PATH)
+
+# Write the data to a new CSV file (no BOM by default)
+xlsx.to_csv(filename=CSV_PATH)
+
+# Optional: explicitly remove BOM if opening a csv that was saved from excel
+# spreadsheet = Roo::Spreadsheet.open("my-csv.csv", { csv_options: { encoding: 'bom|utf-8' } })
+
+header_renames = {
+  "EDG_NUM" => "case_id",
+  "HH Client ID" => "client_id",
+  "HH First Name" => "first_name",
+  "HH Phone Number" => "phone_number",
+  "HH Primary Langauage" => "primary_language",
+}
+csv_data = CSV.read(
+  CSV_PATH,
+  :headers => true,
+  :header_converters => lambda { |header| header_renames[header] || header } # Rename columns
+)
+# Add column preferred_language using column 6 ("HH Primary Langauage")
+csv_data.each do |row|
+  row["preferred_language"] = row[6] == "SPA" ? "es" : "en"
+end
+
+# Add column submit_date
+csv_data.each do |row|
+  row["submit_date"] = row[6] == "SPA" ? "20 de enero" : "January 20"
+end
+
+# Remove columns
+csv_data.delete("CASE_NUM")
+csv_data.delete("HH Last Name")
+csv_data.delete("HH DOB")
+csv_data.delete("TFA")
+csv_data.delete("LTSS")
+
+# Write non-shuffled, pre-chunked data to a place holder csv
+CSV.open(CSV_PATH, "w", :write_headers=> true) do |csv|
+  csv << csv_data.headers
+  csv_data.each{ |row| csv << row }
+end
+
+# Shuffle the file into a random order
+csv_lines = File.readlines(CSV_PATH)
+headers = csv_lines[0]
+content = csv_lines[1..-1]
+content.shuffle!
+
+# Chunk up the file as needed
+chunks = [content[0..0], content[1..10], content[11..110], content[111..1110], content[1111..-1]]
```

Comment:
Minor: we can write a function to compute this dynamically given an input like `[1,10,100, 1000]`.

We should ensure that this does not blow up if there are fewer rows.

Reply:
I'll give this a shot

Reply:
Applied the update if you want to re-review